wagtail/contrib/redirects/tmp_storages.py

```diff
@@ -48,20 +49,12 @@ def remove(self):
 
 
 class TempFolderStorage(BaseStorage):
-    def open(self, mode="r"):
-        if self.name:
-            return open(self.get_full_path(), mode)
-        else:
-            tmp_file = tempfile.NamedTemporaryFile(delete=False)
-            self.name = tmp_file.name
-            return tmp_file
-
     def save(self, data, mode="w"):
-        with self.open(mode=mode) as file:
+        with self._open(mode="w") as file:
```

Comment:
either 
```suggestion
        with self._open(mode="w") as file:
```

or `def save(self)`, otherwise the `mode` kwarg on the save method is superfluous

Reply:
I think keeping this perfectly copied from upstream might be better here. Otherwise the scope for improving it could be so much so that it's not really copied anymore.

Reply:
if which case that should be
https://github.com/django-import-export/django-import-export/blob/ad004e993a5b766596e7954d130253321c77d81c/import_export/tmp_storages.py#L26-L28

Reply:
i added `mode="w"` in the save function as in the utils file `write_to_file_storage` function already gives 2 positional arguments so original implementation causes an error

Reply:
We should definitely err towards modifying Wagtail's code, rather than modifying the code we've copied from `django-import-export`. Not only is it then not a true copy, but it means future upgrades could miss it and introduce regressions.

As mentioned though, I think you need to pass the mode to the constructor, rather than just removing it.